pkg/security/security_profile/profile/manager.go

```diff
@@ -0,0 +1,368 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package profile
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/hashicorp/golang-lru/v2/simplelru"
+	"go.uber.org/atomic"
+
+	"github.com/DataDog/datadog-agent/pkg/security/config"
+	"github.com/DataDog/datadog-agent/pkg/security/metrics"
+	proto "github.com/DataDog/datadog-agent/pkg/security/proto/security_profile/v1"
+	"github.com/DataDog/datadog-agent/pkg/security/resolvers/cgroup"
+	cgroupModel "github.com/DataDog/datadog-agent/pkg/security/resolvers/cgroup/model"
+	"github.com/DataDog/datadog-agent/pkg/security/seclog"
+)
+
+// SecurityProfileManager is used to manage Security Profiles
+type SecurityProfileManager struct {
+	config         *config.Config
+	statsdClient   statsd.ClientInterface
+	cgroupResolver *cgroup.Resolver
+	providers      []Provider
+
+	profilesLock sync.Mutex
+	profiles     map[cgroupModel.WorkloadSelector]*SecurityProfile
+
+	cacheLock sync.Mutex
+	cache     *simplelru.LRU[cgroupModel.WorkloadSelector, *SecurityProfile]
+	cacheHit  *atomic.Uint64
+	cacheMiss *atomic.Uint64
+}
+
+// NewSecurityProfileManager returns a new instance of SecurityProfileManager
+func NewSecurityProfileManager(config *config.Config, statsdClient statsd.ClientInterface, cgroupResolver *cgroup.Resolver) (*SecurityProfileManager, error) {
+	var providers []Provider
+
+	// instantiate directory provider
+	if len(config.SecurityProfileDir) != 0 {
+		dirProvider, err := NewDirectoryProvider(config.SecurityProfileDir, config.SecurityProfileWatchDir)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't instantiate a new security profile directory provider: %w", err)
+		}
+		providers = append(providers, dirProvider)
+	}
+
+	profileCache, err := simplelru.NewLRU[cgroupModel.WorkloadSelector, *SecurityProfile](config.SBOMResolverWorkloadsCacheSize, nil)
```

Comment:
why using `config.SBOMResolverWorkloadsCacheSize`

Reply:
Because it's a mistake 🤦🏻 Nice catch ! 👍🏻